src/wombats/handlers/helpers.clj

```diff
@@ -24,3 +25,105 @@
                                      (not (nil? field-error))
                                      (merge {:field-error field-error}))))))
+
+(defn- format-link
+  [link]
+  (let [query (:query (url link))
+        page-number-string (get query "page")]
+    {:query query
+     :page-number (when page-number-string
+                    (Integer/parseInt page-number-string))
+     :link link}))
+
+(defn- format-links
+  [links]
+  (apply merge
+   (map (fn [link]
+          (let [[ln rel] (clojure.string/split link #";")]
+            (when (and ln rel)
+              {(keyword (last (re-find #"rel=\"(.*)\"" rel)))
+               (format-link (last (re-find #"<(.*)>" link)))})))
+        links)))
+
+(defn parse-link-headers
+  "Parses the link header into a usable data structure.
+
+  Ex: [{:query {}
+        :page-number 0
+        :link \"http://...\"}]
+
+  NOTE: This could be used put into a cljc file and used on the client as well"
+  [context]
+  (let [link-header (get-in context [:request :headers "link"])]
+    (when link-header
+      (-> link-header
+          (clojure.string/split #",")
+          (format-links)))))
+
+(defn- format-url
+  [base uri]
+  (fn [query]
+    (str base uri "?" (clojure.string/join
+                       "&"
+                       (map (fn [[k v]]
+                              (str (name k) "=" v))
+                            query)))))
+
+(defn- generate-link-headers
+  [current-page last-page query url-formatter]
+  (cond-> [{:link (url-formatter (assoc query :page "0"))
+            :rel "first"}
+           {:link (url-formatter (assoc query :page (str last-page)))
+            :rel "last"}]
+    (> current-page 0)
+    (conj {:link (url-formatter (assoc query :page (str (dec current-page))))
+           :rel "prev"})
+
+    (< current-page last-page)
+    (conj {:link (url-formatter (assoc query :page (str (inc current-page))))
+           :rel "next"})))
+
+(defn- format-link-headers
+  [{headers :headers
+    uri :uri
+    query :query-params} current-page last-page]
+  (let [origin (get headers "origin")
+        link-headers (generate-link-headers current-page
+                                            last-page
+                                            query
+                                            (format-url origin uri))]
+    (->> link-headers
+     (map (fn [{:keys [link rel]}]
+            (str "<" link ">; rel=\"" rel "\"")))
+     (clojure.string/join ", "))))
+
+(defn paginate-response
+  "Wraps over a response, paginating the data passed to it. If
+  the data-pred propery is passed, data will be filtered / sorted
+  prior to pagination."
+  [{request :request
+    response :response
+    response-data :response-data
+    page-number-string :page-number
+    per-page :per-page
+    data :data
+    data-pred :data-pred
+    :or {response-data {:status 200}
+         data-pred identity}}]
+
+  (let [page-number (Integer/parseInt (or page-number-string "0"))
```

Comment:
So if the value is passed it will always be a `string` (coming from [pedestal's query param parsing](http://pedestal.github.io/pedestal/io.pedestal.http.route.html#var-parse-query-params)) or will have to default to `"0"` If we try and parse when `page-number-string` is not set we'll get a `NumberFormatException` thrown